docs/hosting/server-setups/digital-ocean.md

```diff
@@ -0,0 +1,178 @@
+# Hosting n8n on DigitalOcean
+
+This hosting guide shows you how to self-host n8n on a DigitalOcean droplet. It uses n8n and [Caddy](http://caddyserver.com){:target="_blank" class=.external-link} (a reverse proxy) to allow access to the Droplet from the internet. It uses [Docker Compose](https://docs.docker.com/compose/){:target="_blank"} to create and define the application components and how they work together.
+
+## Create a Droplet
+
+Log in to DigitalOcean and select the project to host the Droplet and click **Droplets** from the **Manage** menu. You can use [the Docker image](https://marketplace.digitalocean.com/apps/docker){:target="_blank" class=.external-link} from the **Marketplace** tab or use one of the Linux distributions and [install Docker yourself](https://www.docker.com/get-started/){:target="_blank" class=.external-link}. The rest of this tutorial assumes you have Docker installed on the Droplet.
+
+## Log in to your Droplet
+
+The remainder of the steps in this guide require you to log in to the Droplet using a terminal with SSH. [Find more details on how to do this in the Digital Ocean documentation](https://docs.digitalocean.com/products/droplets/how-to/connect-with-ssh/){:target="_blank"}.
+
+## Create folders and files
+
+Both n8n and Caddy require creating folders that the host operating system (the DigitalOcean Droplet) copies to Docker containers to make them available to Docker.
+
+Create the following on the Droplet in your home folder:
+
+- `caddy_config`: Holds the Caddy configuration files.
+- `caddy_data`: A cache folder for Caddy.
+- `local_files`: A folder for files you upload or add via n8n.
+
+Use the following command to create the folders:
+
+```shell
+mkdir caddy_config caddy_data local_files
+```
+
+### Create Docker volume
+
+To persist the Caddy cache between restarts and speed up start times, create [a Docker volume](https://docs.docker.com/storage/volumes/){:target="_blank"} that Docker reuses between restarts.
+
+```shell
+docker volume create caddy_data
+```
+
+## Set up DNS
+
+n8n typically operates on a subdomain. Create a DNS record with your provider for the subdomain and point it to the IP address of the Droplet. The exact steps for this depend on your DNS provider, but typically you need to create a new [CNAME or A record](https://ns1.com/resources/dns-records-explained){:target="_blank"} for the n8n subdomain.
+
+## Open ports
+
+n8n runs as a web application, so the Droplet needs to allow incoming access to traffic on port 80 for non-secure traffic, and port 443 for secure traffic.
+
+Open the following ports in the Droplet's firewall by running the following two commands:
+
+```shell
+sudo ufw allow 80
+sudo ufw allow 443
+```
+
+## Configure n8n
+
+Create an `.env` file in the same folder you will run Docker Compose from, and add the following, replacing the values with your own:
```

Comment:
Have we talked about this before? Might be friendly to specify a directory?

Reply:
If we specify a directory, we'd need to update a lot of other paths in the guide to reflect that. Typically they're found in a project folder or the home folder, which we are using in this case. Personally I'd say keeping the folder structure simpler is possibly more friendly as then there's less to do.

Not quite sure what you mean by "have we talked about this before" though? It's in the current Docker compose tutorial, which is where I took it from. Ideally you don't want to commit sensitive information to a repository and outside of something like Kubernetes, which has an entire mechanism for this, a standard method is to use environment variables.